lib/src/test/kotlin/starknet/provider/response/JsonRpcResponseTest.kt

```diff
@@ -190,4 +189,90 @@ class JsonRpcResponseTest {
         assertEquals("Internal error", exception.message)
         assertEquals("[\"Invalid message selector\",\"0x1234\"]", exception.data)
     }
+
+    @Test
+    fun `rpc provider parses batch callContract`() {
+        val mockResponse = """
+           [
+              {
+                "jsonrpc": "2.0",
+                "result": [
+                  "0x1e2e22799540",
+                  "0x0"
+                ],
+                "id": "0"
+              },
+              {
+                "jsonrpc": "2.0",
+                "result": [
+                  "0x1e2e22799540",
+                  "0x0"
+                ],
+                "id": "1"
+              }
+            ]
+        """.trimIndent()
+
+        val ethContractAddress = Felt.fromHex("0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7")
+
+        val httpServiceMock = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(true, 200, mockResponse)
+        }
+        val provider = JsonRpcProvider("", httpServiceMock)
+
+        val call = Call(
+            ethContractAddress,
```

Comment:
Nit: we can use `DevnetClient.ethErc20ContractAddress` or inline the value since this is a mock test anyway.